---
title: "Infer ASVs with DADA2"
author: "sophiaaredas"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      fig.align = "center",
                      fig.path = "../figures/01_DADA2/") #send any figure output to this folder
```

# Goals of this file

1. Use raw fastq files and generate quality plots.
2. Filter and trim out bad sequences and bases from out sequencing files.
3. Write out fastq files with high quality sequences.
4. Evaluate the quality from our filter and trim.

# Load Libraries
```{r load-libraries}
#install.packages("devtools")
library(devtools)

#devtools::install_github("benjjneb/dada2")
library(dada2)
```

# Load Data
```{r load-data}
# Set the raw fastq path to the raw sequencing files
# Path to the fastq files
raw_fastq_path <- "/workdir/sna49/git_repos/SalinityGradient_16S/data/01_DADA2/01_raw_gizipped_fastqs"
raw_fastq_path

# What files are in this path? Intuition check?
list.files(raw_fastq_path)

# How many files are there?
str(list.files(raw_fastq_path))

# Create a vector of forward reads
forward_reads <- list.files(raw_fastq_path, pattern = "R1_001.fastq.gz", full.names = TRUE)
# final intuition check 
head(forward_reads)

# Create a vector of reverse reads
reverse_reads <- list.files(raw_fastq_path, pattern = "R2_001.fastq.gz", full.names = TRUE)
#final intuition check
head(reverse_reads)
```

# Quality Plots
```{r raw-quality-plot}
# Randomly select 2 samples from dataset to evaluate
random_samples <- sample(1:length(reverse_reads), size = 2)
random_samples

# Calculate and plot quality of these 2 samples
plotQualityProfile(forward_reads[random_samples])

plotQualityProfile(reverse_reads[random_samples])
```

